src/modules/commander/accelerometer_calibration.cpp

```diff
@@ -181,22 +180,21 @@ int do_accel_calibration(int mavlink_fd)
 
 	mavlink_and_console_log_info(mavlink_fd, CAL_QGC_STARTED_MSG, sensor_name);
 
-	struct accel_scale accel_scale = {
-		0.0f,
-		1.0f,
-		0.0f,
-		1.0f,
-		0.0f,
-		1.0f,
-	};
+	struct accel_calibration_s accel_scale;
+	accel_scale.x_offset = 0.0f;
+	accel_scale.x_scale = 1.0f;
+	accel_scale.y_offset = 0.0f;
+	accel_scale.y_scale = 1.0f;
+	accel_scale.z_offset = 0.0f;
+	accel_scale.z_scale = 1.0f;
 
 	int res = OK;
 
 	char str[30];
 
-#ifndef __PX4_QURT
 	/* reset all sensors */
 	for (unsigned s = 0; s < max_accel_sens; s++) {
+#ifndef __PX4_QURT
 		sprintf(str, "%s%u", ACCEL_BASE_DEVICE_PATH, s);
 		/* reset all offsets to zero and all scales to one */
```

Comment:
On QURT, we do reset all the params, just not the ioctl. The use case that you describe works when I tested it.